as_test_yaml_fails because the mimetype is detected as `application/yaml`, which
is not recognized.
diff --git a/libappstream-glib/as-yaml.c b/libappstream-glib/as-yaml.c
index d764d15..da83fea 100644
--- a/libappstream-glib/as-yaml.c
+++ b/libappstream-glib/as-yaml.c
@@ -506,7 +506,8 @@ as_yaml_from_file (GFile *file, AsYamlFromFlags flags, GCancellable *cancellable
 	    g_strcmp0 (content_type, "application/x-gzip") == 0) {
 		conv = G_CONVERTER (g_zlib_decompressor_new (G_ZLIB_COMPRESSOR_FORMAT_GZIP));
 		stream_data = g_converter_input_stream_new (file_stream, conv);
-	} else if (g_strcmp0 (content_type, "application/x-yaml") == 0) {
+	} else if (g_strcmp0 (content_type, "application/x-yaml") == 0 ||
+	    g_strcmp0 (content_type, "application/yaml") == 0) {
 		stream_data = g_object_ref (file_stream);
 	} else {
 		g_set_error (error,
